internal/profiles/validate.go

```diff
@@ -0,0 +1,208 @@
+package profiles
+
+import (
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/pingidentity/pingctl/internal/customtypes"
+	"github.com/spf13/viper"
+)
+
+func Validate() error {
+	// Get a slice of all profile names configured in the config.yaml file
+	profileNames := configProfileNames()
+
+	// Validate profile names
+	if err := validateProfileNames(profileNames); err != nil {
+		return err
+	}
+
+	// for each profile key, set the profile based on mainViper.Sub() and validate the profile
+	for _, name := range profileNames {
+		v := mainViper.Sub(name)
+
+		if err := validateProfileKeys(name, v); err != nil {
+			return err
+		}
+
+		if err := validateProfileValues(v); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateProfileNames(profileKeys []string) error {
+	for _, profileKey := range profileKeys {
+		re := regexp.MustCompile(`^[a-zA-Z0-9\_\-]+$`)
+		if !re.MatchString(profileKey) {
+			return fmt.Errorf("failed to validate pingctl configuration: profile name '%s' must contain only alphanumeric characters, underscores, and dashes", profileKey)
+		}
+	}
+	return nil
+}
+
+func configProfileNames() []string {
+	allKeys := mainViper.AllKeys()
+
+	// Get a slice of all profile keys to validate profiles individually
+	var profileKeys []string
+	for _, key := range allKeys {
+		//remove "activeProfile" from profileKeys
+		if strings.EqualFold(key, ProfileOption.ViperKey) {
+			continue
+		}
+
+		rootKey := strings.Split(key, ".")[0]
+		if !slices.Contains(profileKeys, rootKey) {
```

Comment:
Might be more trouble than it's worth, but you can use a `map[string]whatever` as a set rather than running a contains on a slice every time

Reply:
Given the function is used once, and for readability, I will keep the contains usage.